File: src/backend/Services/Reservations/OrangeCarRental.Reservations.Api/Contracts/CustomerDetailsDto.cs
namespace SmartSolutionsLab.OrangeCarRental.Reservations.Api.Contracts;

/// <summary>
///     Customer details for inline registration.
/// </summary>
public sealed record CustomerDetailsDto(
    string FirstName,
    string LastName,
    string Email,
    string PhoneNumber,
    DateOnly DateOfBirth);

File: src/frontend/libs/reservation-api/src/lib/models/customer-details.dto.ts
/**
 * Customer details
 */
import type {
  ISODateString,
  EmailAddress,
  PhoneNumber,
  FirstName,
  LastName,
} from "@orange-car-rental/shared";

export type CustomerDetails = {
  readonly firstName: FirstName;
  readonly lastName: LastName;
  readonly email: EmailAddress;
  readonly phoneNumber: PhoneNumber;
  readonly dateOfBirth: ISODateString;
};
